src/main/java/net/minecraftforge/common/loot/LootRegistry.java

```diff
@@ -0,0 +1,89 @@
+package net.minecraftforge.common.loot;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.storage.loot.LootTable;
+import net.minecraftforge.fml.common.FMLLog;
+import net.minecraftforge.fml.common.Loader;
+import net.minecraftforge.fml.common.LoaderState;
+import net.minecraftforge.fml.common.ModContainer;
+
+import java.io.BufferedInputStream;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Map;
+
+public class LootRegistry {
+
+    /**
+     * This is called by LootTableManager after it looks for loot tables in the world folder
+     * and before it looks for loot tables in vanilla.
+     * By default, loot tables are loaded from the resource path (jar) of the mod whose modid matches the resource domain of the given path.
+     * If there exists an override handler, then that is used instead.
+     * @param path The path of the loot table to load.
+     * @param deserializer The gson that can handle Loot Tables.
+     * @return A LootTable object, or null indicating that the LootTableManager should continue looking elsewhere.
+     */
+    public static LootTable loadModLootTable(ResourceLocation path, Gson deserializer)
+    {
+        String modid = path.getResourceDomain();
+        InputStream input = null;
+
+        if (LOADING_OVERRIDES.containsKey(modid))
+        {
+            input = LOADING_OVERRIDES.get(modid).apply(path);
+        }
+        else
+        {
+            ModContainer mod = Loader.instance().getIndexedModList().get(modid);
+            if (mod != null)
+            {
+                input = mod.getMod().getClass().getResourceAsStream("/assets/" + path.getResourceDomain() + "/loot_tables/" + path.getResourcePath() + ".json");
+            }
+        }
+
+
+        if (input == null)
+        {
+            return null;
+        }
+
+        try
+        {
+            return deserializer.fromJson(new InputStreamReader(new BufferedInputStream(input)), LootTable.class);
```

Comment:
We need to use a custom deserializer that is more strict then vanilla's, and has more features.
every entry MUST have a "id" field, EntryItem can be defaulted to item_name IF it's only one entry for that item. Else throw a malformed error or something.
EntryTable, again, the same for that their ID can be the table's resource location. 

Forcing an ID allows for people to do things like LootTables.get("minecraft:fishing").remove("main.minecraft:fishing_treasure") to remove all treasures from the possible fishing loot.